File: Src2/Icm.Core/Collections extensions/ICollectionExtensions.vb
Imports System.Text
Imports System.Runtime.CompilerServices

Namespace Icm.Collections

    Public Module ICollectionExtensions

        ''' <summary>
        ''' Remove an item, not failing if it doesn't exist.
        ''' </summary>
        ''' <typeparam name="T"></typeparam>
        ''' <param name="c"></param>
        ''' <param name="item"></param>
        ''' <remarks></remarks>
        <Extension()>
        Sub ForceRemove(Of T)(ByVal c As ICollection(Of T), ByVal item As T)
            If c.Contains(item) Then
                c.Remove(item)
            End If
        End Sub

    End Module

End Namespace

File: Src/Icm.Core/Collections extensions/ICollectionExtensions.cs
using System.Collections.Generic;
using System.Runtime.CompilerServices;

namespace Icm.Collections
{

	public static class ICollectionExtensions
	{

		/// <summary>
		/// Remove an item, not failing if it doesn't exist.
		/// </summary>
		/// <typeparam name="T"></typeparam>
		/// <param name="c"></param>
		/// <param name="item"></param>
		/// <remarks></remarks>
		[Extension()]
		public static void ForceRemove<T>(ICollection<T> c, T item)
		{
			if (c.Contains(item)) {
				c.Remove(item);
			}
		}

	}

}

//=======================================================
//Service provided by Telerik (www.telerik.com)
//Conversion powered by NRefactory.
//Twitter: @telerik
//Facebook: facebook.com/telerik
//=======================================================
